tests/integration/upgrade/test_upgrade.py

```diff
@@ -31,12 +34,41 @@ async def test_build_and_deploy(ops_test: OpsTest) -> None:
     if app_name:
         return await check_or_scale_app(ops_test, app_name)
 
-    my_charm = await ops_test.build_charm(".")
-    await ops_test.model.deploy(my_charm, num_units=3)
-    await ops_test.model.wait_for_idle()
+    await ops_test.model.deploy("mongodb", channel="stable", num_units=3)
+    await ops_test.model.wait_for_idle(
+        apps=["mongodb"], status="active", timeout=1000, idle_period=120
+    )
+
+
+@pytest.mark.group(1)
+@pytest.mark.skip(reason="Must wait for stable channel to support upgrades before testing")
+async def test_upgrade(ops_test: OpsTest, continuous_writes) -> None:
+    """Verifies that the upgrade can run successfully."""
+    app_name = await get_app_name(ops_test)
+    leader_unit = await find_unit(ops_test, leader=True, app_name=app_name)
+    logger.info("Calling pre-upgrade-check")
+    action = await leader_unit.run_action("pre-upgrade-check")
+    await action.wait()
+
+    await ops_test.model.wait_for_idle(
+        apps=[app_name], status="active", timeout=1000, idle_period=120
+    )
+
+    new_charm = await ops_test.build_charm(".")
+    app_name = await get_app_name(ops_test)
+    await ops_test.model.applications[app_name].refresh(path=new_charm)
+    await ops_test.model.wait_for_idle(
+        apps=[app_name], status="active", timeout=1000, idle_period=120
+    )
+
```

Comment:
I believe we should call `test_consistency_between_workload_and_metadata` here

Reply:
`test_consistency_between_workload_and_metadata` is a sanity check that tests that:

when the developer updates the snap revision, that they should have upgraded the version in the variable DEPENDENCIES. If they do not then they will be mismatched and upgrades will not be checking feasible revisions properly. 

I know this is kind of confusing, so if a call would be better let me know

Reply:
Sorry, I misread that function for a helper. Please disregard